SConstruct

```diff
@@ -334,6 +329,16 @@ if selected_platform == "javascript":
     print('Platform "javascript" has been renamed to "web" in Godot 4. Building for platform "web".')
     selected_platform = "web"
 
+if selected_platform not in platform_list:
+    if selected_platform == "":
+        print("Could not detect platform automatically.")
+    elif selected_platform != "list":
+        print(f'Invalid target platform "{selected_platform}".')
+
+    print(f"The following platforms are available:\n\t{'\n\t'.join(platform_list)}\n")
```

Comment:
For the record, this doesn't seem to be legal in Python 3.10. So I fixed it up in 7e4c150573d6af7072f2b55ae76dc7b723f21f66 (and IMO that's more readable anyway ;)).